app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,51 @@
+class Api::V1::UsersController < ApplicationController
+    before_action :set_user, only: [:show, :update, :destroy]
+
+    # GET /users
+    def index
+      @users = User.all
+      render json: @users
+    end
+  
+    # GET /users/:id
+    def show
+      render json: @user, status: 200
+    # your code goes here
+    end
+  
+    # POST /users
+    def create
+      @user = User.new(user_params)
+      if @user.save
+        render json: @user, status: 201
+      else
+        render json: { error:
+          "Unable to create user: #{@user.errors.full_messages.to_sentence}"},
+          status: 400
+      end
+    end
+  
+    # PUT /users/:id
+    def update
+      @user.update(user_params)
+      flash.notice = "The user record was created successfully."
```

Comment:
This flash is unnecessary as well and should be removed so it does not cause errors.